docs/parcels.md

```diff
@@ -0,0 +1,149 @@
+# Parcels
+
+A single-spa parcel is a bundle of functionality (like an application or component) meant to be mounted manually 
+by an application. Parcels use similar methodology as applications but are mounted by a manual function call rather than the activity function.
+A parcel can be as large as an application or as small as an component and written in 
+any language as long as it exports the correct lifecycle events. In a single-spa world, your SPA contains 
+many registered applications and potentially many parcels. Typically we recommend you mount a parcel within 
+the context of an application because the parcel will be unmounted with the application.
+
+## Parcel configuration
```

Comment:
Most people seem to want to see the simplest code example possible almost immediately, before diving into the concepts and deeper documentation. Maybe before this section we could put a very simple example?

> ## Example
> ```js
> // The parcel implementation
> const parcelConfig = {
>   bootstrap() {
>     // one time initialization
>     return Promise.resolve()
>   },
>   mount() {
>     // use a framework to create dom nodes and mount the parcel
>     return Promise.resolve()
>   },
>   unmount() {
>     // use a framework to unmount dom nodes and perform other cleanup
>     return Promise.resolve()
>   }
> }
>
> // How to mount the parcel
> const domElement = document.getElementById('place-in-dom-to-mount-parcel')
> const parcelProps = {domElement, customProp1: 'foo'}
> const parcel = singleSpa.mountParcel(parcelConfig, parcelProps)
>
> // The parcel is being mounted. We can wait for it to finish with the mountPromise.
> parcel.mountPromise.then(() => {
>   console.log('finished mounting parcel!')
>   // If we want to re-render the parcel, we can call the update lifecycle method, which returns a promise
>   parcelProps.customProp1 = 'bar'
>   return parcel.update(parcelProps)
> })
> .then(() => {
>   // Call the unmount lifecycle when we need the parcel to unmount. This function also returns a promise
>   return parcel.unmount()
> })
> ```